### PR TITLE
[excel] (Comments) Adding section to workbook article about comment APIs

### PR DESCRIPTION
**Pull request by AlexJerabek**
This section describes how to add, edit, and delete comments and comment replies. It also corrects some of the method references in the workbook article to be more consistent.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[ef0d825](https://github.com/OfficeDev/office-js-docs-pr/pull/911/commits/ef0d82521c0f7fadc262329d68a38b130742147e)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C22%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904221834483671-911%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=JkOrwIyB2M3XzY2RzofvOqua7sR0xDKWmybRMfDJZnQ%3D&st=2019-04-22T18%3A30%3A54Z&se=2019-05-23T18%3A35%3A54Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-workbooks.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Work with workbooks using the Excel JavaScript API
 description: ''
-ms.date: 02/28/2019
+ms.date: 04/22/2019
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: Due to my tardiness on reviewing this, we probably should update the ms.date value.

File: docs/excel/excel-add-ins-workbooks.md [hunk 2]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
 context.application.suspendApiCalculationUntilNextSync();
 ```
 
+## Comments
+
+> [!NOTE]
+> The comment APIs are currently available only in public preview. [!INCLUDE [Information about using preview APIs](../includes/using-excel-preview-apis.md)]
+
+All the [comments](https://support.office.com/article/insert-comments-and-notes-in-excel-bdcc9f5d-38e2-45b4-9a92-0b2b5c7bf6f8) added to a workbook are tracked by the `Workbook.comments` property. This is a [CommentCollection](/javascript/api/excel/excel.commentcollection) object, which maintains a collection of [Comment](/javascript/api/excel/excel.comment) objects. This collection contains comments created by the user and comments created by your add-in.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: Consider rephrasing ->

All [comments](https://support.office.com/article/insert-comments-and-notes-in-excel-bdcc9f5d-38e2-45b4-9a92-0b2b5c7bf6f8) within a workbook are tracked by the `Workbook.comments` property, including comments created by users and also comments created by your add-in. The `Workbook.comments` property is a [CommentCollection](/javascript/api/excel/excel.commentcollection) object that contains a collection of [Comment](/javascript/api/excel/excel.comment) objects.

File: docs/excel/excel-add-ins-workbooks.md [hunk 3]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
...
+
+Comments are added to the workbook with the `CommentCollection.add` method. This method takes in the comment's text, as a string, and the cell where the comment will go, as a string or [Range](/javascript/api/excel/excel.range) object. The following code sample adds a comment to cell **A2**.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: "Comments are added to the workbook..." sounds passive -- consider rewording to be more active voice. For example:
- You can add comments to a workbook by using the `CommentCollection.add` method.
  OR
- To add comments to a workbook, use the `CommentCollection.add` method.

Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: Consider changing "where the comment will go" -> "where the comment should be added"

Comment (kbrandl) [reply docs/excel/excel-add-ins-workbooks.md]: Actually, you might consider combining the first two sentences, something like ->

You can add comments to a workbook by using the `CommentCollection.add` method, passing in the comment's text, as string, and the cell where the comment should be added, as either a string or [Range](/javascript/api/excel/excel.range) object.

File: docs/excel/excel-add-ins-workbooks.md [hunk 4]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
...
+Excel.run(function (context) {
+    var comments = context.workbook.comments;
+
+    // Note that an InvalidArgument error will be thrown if multiple cells passed to `Comment.add`.
+    comments.add("TODO: add data.", "A2");
+    return context.sync();
+});
+```
+
+Comments contain metadata about their creation, such as the author and creation date. Comments created by your add-in are considered to be authored by the current user. The following sample shows how to display the author's email, author's name, and creation date of a comment at **A2**.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: "Comments contain metadata about their creation, such as the author and creation date."  -> 

"Each comment contains metadata about its creation, such as author and creation date."

File: docs/excel/excel-add-ins-workbooks.md [hunk 5]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
...
+    // Get the comment at cell A2.
+    var comment = context.workbook.comments.getItemByCell("Comments!A2");
+    comment.load(["authorEmail", "authorName", "creationDate"]);
+    return context.sync().then(function () {
+        console.log(`${comment.creationDate.toDateString()}: ${comment.authorName} (${comment.authorEmail})`);
+    });
+});
+```
+
+Every comment forms the head of a thread containing zero or more replies. `Comment` objects have a `replies` property, which is a [CommentReplyCollection](/javascript/api/excel/excel.commentreplycollection) that contains [CommentReply]/javascript/api/excel/excel.commentreply) objects. `CommentReplyCollection.add` creates a new reply using the provided content string. Replies are displayed in the order they are added. The following code sample adds a reply to the first comment in the workbook.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: "Every comment forms the head of a thread containing zero or more replies" -- this sentence sounds somewhat confusing, IMO. Perhaps consider rewording to -> 

Each comment may contain zero or more replies.

Comment (kbrandl) [reply docs/excel/excel-add-ins-workbooks.md]: The markdown for the `CommentReply` is malformed -- you're missing the opening parenthesis before the URL/path.

Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: Suggest rephrasing the sentence that begins with `CommentReplyCollection.add` ->  

To add a reply to a comment, use the `CommentReplyCollection.add` method, passing in the text of the reply.

File: docs/excel/excel-add-ins-workbooks.md [hunk 6]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
...
+```js
+Excel.run(function (context) {
+    // Get the first comment added to the workbook.
+    var comment = context.workbook.comments.getItemAt(0);
+    comment.replies.add("Thanks for the reminder!");
+    return context.sync();
+});
+```
+
+Comments and comment replies can be edited by setting their `Comment.content` or `CommentReply.content` properties. They can be deleted using the `Comment.delete` or `CommentReply.delete` methods. Deleting a comment also deletes all the replies associated with that comment.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: Consider rewording to use more active voice ->

To edit a comment or comment reply, set its `Comment.content` property or `CommentReply.content` property. To delete a comment or comment reply, use the `Comment.delete` method or `CommentReply.delete` method.

File: docs/excel/excel-add-ins-workbooks.md (hunk 5 above)
Comment (AlexJerabek) [reply docs/excel/excel-add-ins-workbooks.md]: @kbrandl Is it misleading to say "To reply to a comment..."?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[4b151de](https://github.com/OfficeDev/office-js-docs-pr/pull/911/commits/4b151de2c897a448d61935d4e982e710909eb374)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C29%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904291814413022-911%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=kr8DA897jzONCee8M%2Bn4Uhaoq1Mkc3t2uyN5XFdB1cM%3D&st=2019-04-29T18%3A10%3A50Z&se=2019-05-30T18%3A15%3A50Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by AlexJerabek** [comment]
Thanks @kbrandl! I've incorporated your feedback. I did have a question: is it misleading to describe the add-in's action as "replying to a comment" instead of "adding a reply to a comment"? The former sounds better to me, but may not be technically correct.

File: docs/excel/excel-add-ins-workbooks.md [hunk 7]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
...
+    // Get the comment at cell A2.
+    var comment = context.workbook.comments.getItemByCell("Comments!A2");
+    comment.load(["authorEmail", "authorName", "creationDate"]);
+    return context.sync().then(function () {
+        console.log(`${comment.creationDate.toDateString()}: ${comment.authorName} (${comment.authorEmail})`);
+    });
+});
+```
+
+Each comment contains zero or more replies. `Comment` objects have a `replies` property, which is a [CommentReplyCollection](/javascript/api/excel/excel.commentreplycollection) that contains [CommentReply](/javascript/api/excel/excel.commentreply) objects. To add a reply to a comment, use the `CommentReplyCollection.add`method, passing in the text of the reply. Replies are displayed in the order they are added. The following code sample adds a reply to the first comment in the workbook.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: You're missing a space between  `CommentReplyCollection.add` and the word "method" in "... use the `CommentReplyCollection.add`method,"

File: docs/excel/excel-add-ins-workbooks.md [hunk 8]
```diff
@@ -257,12 +257,60 @@ The Excel API also lets add-ins turn off calculations until `RequestContext.sync
...
+```js
+Excel.run(function (context) {
+    // Get the first comment added to the workbook.
+    var comment = context.workbook.comments.getItemAt(0);
+    comment.replies.add("Thanks for the reminder!");
+    return context.sync();
+});
+```
+
+TO edit a comment or comment reply, set its `Comment.content` property or `CommentReply.content` property. To delete a comment or comment reply, use the `Comment.delete` method or `CommentReply.delete` method. Deleting a comment also deletes all the replies associated with that comment.
```
Comment (kbrandl) [inline docs/excel/excel-add-ins-workbooks.md]: "TO" at the start of the sentence should be "To"

**Comment by kbrandl** [comment]
@AlexJerabek re your question about "replying to a comment" versus "adding a reply to a comment" --

IMO, although it's wordier, "To add a reply to a comment" is more accurate in this context than "To reply to a comment".   When describing an add-in's action, it should be done in terms of CRUD -- 
* Create (add)
* Read (get)
* Update
* Delete

**Comment by o365devx** [comment]
Docs Build status updates of commit _[3c5318c](https://github.com/OfficeDev/office-js-docs-pr/pull/911/commits/3c5318c81b8f94bf560eeda80595763436a8c78e)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C29%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201904291839558528-911%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=9NF19yPLoehJzwlQhXVAJarkLQHfbRCIaZAC9ejWPCQ%3D&st=2019-04-29T18%3A36%3A04Z&se=2019-05-30T18%3A41%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**